frontend/src/components/ServiceForm/ServiceForm.tsx

```diff
@@ -45,6 +52,17 @@ export const ServiceForm: React.FC<Props> = ({ service, target = DEFAULT_TARGET,
   const appType = findType(applicationTypes, form.type)
   const css = useStyles()
 
+  useEffect(() => {
+    if ((form?.port, form?.hostname)) {
+      serviceDispatch.checkService({
```

Comment:
I don't think we need a service model. It's such a simple set and emit I think we should use the backend model like we do for the freeport scan.